app/routes/news.$category.tsx

```diff
@@ -0,0 +1,42 @@
+import { invariant } from '@epic-web/invariant'
+import { type LoaderFunctionArgs, json } from '@remix-run/node'
+import { useLoaderData } from '@remix-run/react'
+import { toTitleCase } from '~/utils/stringUtils.ts'
+
+export async function loader({ params }: LoaderFunctionArgs) {
+	const { category } = params
+
+	invariant(typeof category === 'string', 'Category not found')
+	const categoryTitle = toTitleCase(category)
+
+	return json({ categoryTitle })
+}
+
+const WireframeBlock = () => {
+	return (
+		<div className="h-72 animate-pulse rounded-lg bg-gray-200 dark:bg-gray-700" />
+	)
+}
+
+export default function NewsCategoryPage() {
+	const { categoryTitle } = useLoaderData<typeof loader>()
+
+	return (
+		<div className="container py-16">
+			<h2 className="text-h2">{categoryTitle}</h2>
+
+			<div className="grid grid-cols-5 gap-6 p-8">
+				<WireframeBlock />
+				<WireframeBlock />
+				<WireframeBlock />
+				<WireframeBlock />
+				<WireframeBlock />
+				<WireframeBlock />
+				<WireframeBlock />
+				<WireframeBlock />
+				<WireframeBlock />
+				<WireframeBlock />
```

Comment:
You have made promising progress here, Tyler. Users can add their own news stories and attach images to them.

You now need to [load and display article data](https://gedsargent.github.io/code-monster/guides/epic-news/displaying-articles/load-and-display-article-data/) in place of the wireframe blocks above.

Once this is finished, you still need to create a single article page, and then a functioning admin area. This is a lot of work still to do, so you will need to pick up the pace of your work.